StackReviewSuggestedEditsRework.user.js

```diff
@@ -2185,22 +2185,14 @@
             topBar.append(settingsItem);
 
             // loads the GUI for the settings
-            settingsItem.addEventListener("click", (event) => {
-                const sleep = (milliseconds) => {
-                    return new Promise(resolve => setTimeout(resolve, milliseconds));
+            settingsItem.addEventListener("click", () => {
+                let modalElement = document.querySelector(`#${modalConfig.ids.aside}`);
+                if (!modalElement) { // check if the modal has been appended to the DOM
+                    modalElement = createModalAside(loadIt());
+                    document.body.appendChild(modalElement);
                 }
-
-                const aside = createModalAside(loadIt());
-                settingsItem.append(aside);
-                settingsLink.dataset.action = show;
-                settingsItem.dataset.controller = modal;
-
-                // The GUI needs to load first..
-                sleep(50).then(() => {
-                    settingsLink.click();
-                    // Stacks.showModal(settingsItem);
-                })
-            }, { once: true });
+                setTimeout(() => Stacks.showModal(modalElement), 0);
```

Comment:
btw, can somebody explain what the heck are we doing there? I mean I get this is a timeout scheduled for the "immediate" execution, but maybe we have a better way to determine if everything has loaded?

Reply:
No idea why `Stacks.showModal()` doesn't work, but `StackExchange.helpers.showModal();` for some reason does. Do you think we could use the latter or load the modal for the first time on hover?

Reply:
> No idea why `Stacks.showModal()` doesn't work, but `StackExchange.helpers.showModal();` for some reason does. Do you think we could use the latter or load the modal for the first time on hover?

mm, something tells me there is a bug in `showModal`. Take a look at [this line](https://github.com/StackExchange/Stacks/blob/8b49250a97b3da3f6be31b6de0f843697388c97b/lib/ts/controllers/s-modal.ts#L320). The comment says it "toggles the modal if left undefined" but due to how they check for truthiness, the modal always gets hidden. Maybe this is the reason?

Reply:
Hmm... With the `StackExchange` method, if I click the gear icon, then `x` or Cancel and then again the gear icon, the modal opens for a moment and then it's hidden/removed again.